requirements/dev.txt

```diff
@@ -2,82 +2,93 @@
+#    pip-compile --no-index --output-file requirements/dev.txt requirements/base.in requirements/dev.in requirements/quality.in
 #
-
 amqp==1.4.9               # via kombu
 anyjson==0.3.3            # via kombu
 appdirs==1.4.3            # via fs
-argparse==1.4.0           # via caniusepython3
-astroid==1.5.2            # via edx-lint, pylint, pylint-celery
-backports.functools-lru-cache==1.5  # via caniusepython3
+astroid==1.5.3            # via edx-lint, pylint, pylint-celery
 billiard==3.3.0.23        # via celery
-caniusepython3==7.0.0
+bleach==3.0.2             # via readme-renderer
 celery==3.1.18
-certifi==2018.4.16        # via requests
+certifi==2018.10.15       # via requests
 chardet==3.0.4            # via requests
 click-log==0.1.8          # via edx-lint
-click==6.7                # via click-log, edx-lint, pip-tools
-diff-cover==1.0.4
-distlib==0.2.7            # via caniusepython3
+click==7.0                # via click-log, edx-lint, pip-tools
+diff-cover==1.0.5
 django-braces==1.13.0     # via django-oauth-toolkit
 django-model-utils==3.1.2
 django-oauth-toolkit==0.12.0
+django-waffle==0.14.0     # via edx-django-utils, edx-drf-extensions
 django==1.10.8
+djangorestframework-jwt==1.11.0  # via edx-drf-extensions
 djangorestframework==3.6.4
+docutils==0.14            # via readme-renderer
 edx-celeryutils==0.1.5
-edx-completion==0.1.8
+edx-completion==0.1.14
+edx-django-utils==1.0.1   # via edx-drf-extensions
+edx-drf-extensions==1.11.0  # via edx-completion
 edx-i18n-tools==0.4.7
-edx-lint==0.5.5
+edx-lint==0.6.0
 edx-opaque-keys[django]==0.4.4
-first==2.0.1              # via pip-tools
-fs==2.0.24                # via xblock
+filelock==3.0.10          # via tox
+fs==2.1.1                 # via xblock
+future==0.17.1            # via pyjwkest
 idna==2.7                 # via requests
-inflect==0.3.1            # via jinja2-pluralize
+importlib-metadata==0.6   # via path.py
+inflect==1.0.1            # via jinja2-pluralize
 isort==4.3.4
 jinja2-pluralize==0.3.0   # via diff-cover
 jinja2==2.10              # via diff-cover, jinja2-pluralize
 jsonfield==2.0.2          # via edx-celeryutils
 kombu==3.0.37             # via celery
 lazy-object-proxy==1.3.1  # via astroid
-lxml==4.2.3               # via xblock
+lxml==4.2.5               # via xblock
 markupsafe==1.0           # via jinja2, xblock
 mccabe==0.6.1             # via pylint
 mysqlclient==1.3.13
+newrelic==4.4.1.104       # via edx-django-utils
 oauthlib==2.0.1           # via django-oauth-toolkit
-packaging==17.1           # via caniusepython3, tox
-path.py==11.0.1           # via edx-i18n-tools
-pbr==4.1.0                # via stevedore
-pip-tools==2.0.2
+path.py==11.5.0           # via edx-i18n-tools
+pbr==5.1.0                # via stevedore
+pip-tools==3.1.0
 pkginfo==1.4.2            # via twine
-pluggy==0.6.0             # via tox
+pluggy==0.8.0             # via tox
```

Comment:
Make requirements failed for me because this file wants pluggy 0.8 while test.txt wants 0.6. 
Switching to pytest==3.7.x should resolve this I think.

Reply:
So switching to a newer pytest helps this issue, but creates a new issue in that the test suite starts complaining about django.conf.urlresolvers being deprecated, but we need to use it to continue to support django 1.8.  I'll see if I can silence those warnings with a conditional import, and then upgrade pytest again.

Reply:
Resolving that lead to a complaint about pytest fixtures.  And then on to a complaint about not being able to find contextlib (only needed for python 2).  I tried building all the requirements from python 2 instead of 3, but that lead to another set of errors.